apis/configwatch_test.go

```diff
@@ -0,0 +1,119 @@
+// Copyright Contributors to the L3AF Project.
+// SPDX-License-Identifier: Apache-2.0
+
+package apis
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestMatchHostnamesWithRegexp(t *testing.T) {
+	type args struct {
+		pattern string
+		host    string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    bool
+		wantErr bool
+	}{
+		{
+			name:    "EmptypCheck",
+			args:    args{pattern: "", host: ""},
+			want:    false,
+			wantErr: false,
+		},
+		{
+			name:    "LengthMissMatchCheck",
+			args:    args{pattern: "l3afd-lfn.us.l3af.io", host: "l3afd-lfn.l3af.io"},
```

Comment:
We should probably rename the args to be more obvious.  Pattern is really DNS Name here and host is the regex.